analyzer/src/main/kotlin/managers/Npm.kt

```diff
@@ -240,7 +244,35 @@ open class Npm(
                     .url("https://registry.npmjs.org/$encodedName")
                     .build()
 
-                OkHttpClientHelper.execute(HTTP_CACHE_PATH, pkgRequest).use { response ->
+                OkHttpClientHelper.execute(HTTP_CACHE_PATH, pkgRequest) {
+                    val npmrcFile = getUserHomeDirectory().resolve(".npmrc")
+                    if (npmrcFile.isFile) {
+                        val httpRegex = Regex("^https?://.+$")
+
+                        npmrcFile.forEachLine { line ->
+                            val (key, value) = line.split('=', limit = 2).map { it.trim() }
+
+                            val proxyUrl = value.takeIf { it.matches(httpRegex) } ?: when (key) {
+                                "proxy" -> "http://$value"
+                                "https-proxy" -> "https://$value"
+                                else -> ""
+                            }
```

Comment:
Would be nice to extract the part that reads the proxy URL from `.npmrc` to a separate function.

Reply:
Done.